stagehand/src/mcp-sampling-client.ts

```diff
@@ -0,0 +1,275 @@
+import { Server } from "@modelcontextprotocol/sdk/server/index.js";
+import {
+  CreateMessageRequestSchema,
+  CreateMessageResultSchema,
+} from "@modelcontextprotocol/sdk/types.js";
+import { 
+  LLMClient, 
+  CreateChatCompletionOptions, 
+  LLMResponse, 
+  ChatMessage,
+  AvailableModel
+} from "@browserbasehq/stagehand";
+
+/**
+ * LLM Client that uses MCP Sampling to request completions from the MCP client
+ * instead of using a direct API key to an LLM provider.
+ */
+export class MCPSamplingLLMClient extends LLMClient {
+  private server: Server;
+
+  constructor(server: Server, modelName: AvailableModel = "gpt-4o", userProvidedInstructions?: string) {
```

Comment:
Instead of hardcoding the available model, does the MCP Sampling protocol allow for the client to specify which of the available models will be made available to the MCP Server?